File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        freq = {}
        max = 0
        val = 0
        for num in nums:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
            if freq[num]>max:
                max = freq[num]
                val = num
        return val
            

File: 0169-majority-element/0169-majority-element.cpp
#include <unordered_map>
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> hashmap; // num, freq
        int n = nums.size();
        for (int i=0; i<n; i+=1){
            hashmap[nums[i]]++;
            if (hashmap[nums[i]]>(n/2)) return nums[i];
        }
        return -1;     
    }
};